### ISSUE TITLE
binary releases are always way behind git tags

### ISSUE THREAD

**Issue by gracenoah** [zcash/zcash#4127]
### Describe the issue
We, and probably many others rely on binary releases rather than building our dependencies from scratch. However, zcash binary releases have been very unreliable. A release of 2.0.7-1 was never published. Its git tag was removed when 2.0.7-2 was released and 2.0.7-2 is not published either.

#### If so, please list the steps to reproduce below:
1. Wait for binary releases

### Expected behaviour
A binary release is published the same day as the github "release"

### Actual behaviour + errors
Waiting for days, weeks, undefined amounts of time, not sure if or when a binary release will be made.

### The version of Zcash you were using:
2.0.7

### Machine specs:
n/a

### Any extra information that might be useful in the debugging process.
This has happened for all releases I've seen so far.

### Do you have a backup of `~/.zcash` directory and/or take a VM snapshot?
n/a

If we can't have binary releases in a timely fashion, I honestly would prefer no binary releases at all so that everyone knows that they have to build from source themselves. I don't mind building from source. I do mind not knowing what to expect. The delay of binary releases combined with an aggressive hard fork schedule makes it very scary to rely on them. Please either provide some clarity about what we should expect so we don't end up being hard forked off.

**Comment by daira** [comment]
The binary release is typically available on the first working day after the tag. We've had a couple of releases recently when the tag was done late on Friday or at a weekend, and this time Monday was an official holiday in the US. That's all. There's is no point in delaying the tag just because the binary release isn't done.

**Comment by gracenoah** [comment]
1. I remember having to wait weeks for either the binary or the debian package to be published after a previous release. I don't remember if it was 2.0.6 or 2.0.7. I didn't record every instance of delays, obviously, but I've had to modify our build system to use either the binary release or the debian package release based on which one becomes available first because only one or the other was available at different times in the past.

2. It was very unclear that there was a problem with v2.0.7-1. The only thing I saw was that the binaries were not available and then, a week later, the git tag disappered. If we had changed our release process to build from scratch due to 1., we would have picked up the bad release. Where and how was the warning to not deploy v2.0.7-1 published?

**Comment by daira** [comment]
You're right that we failed to communicate sufficiently about v2.0.7-1. That was my fault as release manager, and I apologize. The binaries for v2.0.7-2 will be available today.

**Comment by daira** [comment]
v2.0.7-2 has been released: https://electriccoin.co/blog/new-release-2-0-7-2/

**Comment by gracenoah** [comment]
Thanks!

Just to confirm I wasn't imagining things, I compared the blog post dates with the git tag dates:

| version | blog       | tag        | diff |
|---------|------------|------------|------|
| 2.0.7-2 | 2019-09-04 | 2019-09-02 | 2    |
| 2.0.7   | 2019-08-26 | 2019-08-20 | 6    |
| 2.0.6   | 2019-07-15 | 2019-07-01 | 14   |
| 2.0.5-2 | 2019-05-21 | 2019-05-22 | -1    |
| 2.0.5-1 | n/a        | 2019-05-22 | n/a  |
| 2.0.5   | n/a        | 2019-05-22 | n/a  |
| 2.0.4   | 2019-03-27 | 2019-03-27 | 0    |

I don't have data about when packages or binaries were available. I don't understand what happened in 2.0.5, but I assume binaries were available after the tag was created, which means that actually, the blog post was made before the release was done.

I guess my ask is just the following: Tell me how to be notified when I can actually upgrade our node to the latest version so I don't have to keep checking every day and wondering what's going on. I'm very flexible. I can subscribe to an RSS feed, mailing list, git tags, blog posts, I can even set up a scraper for website changes. I just want to not have to guess how every release will be announced and when. Even if the only option is to poll for the n+1'th download via templating the download URL or poll the debian repo for updates, that's fine, but I have to know so I can set up a notification system.

**Comment by gracenoah** [comment]
I noticed that 2.0.7-3 was tagged and the binaries were released all within the same day! Thanks for that!

Looks like the release was not announced in the usual place though: https://electriccoin.co/blog/category/technical/

I didn't know about https://z.cash/support/security/announcements/ until now. Is there a better way to get notifications than setting up a web scraper to periodically check for changes to the page?
